PROCESSES.md

```diff
@@ -43,6 +43,7 @@ When promoting a chart from incubator to stable there are several steps that nee
 1. The chart should be copied, not moved, from the incubator directory to the stable directory.
 1. The chart in the incubator directory should be deprecated according to the [deprecation process](#deprecating-a-chart) described above with a comment noting that the chart has been promoted to stable.
 1. The version of the chart in the stable directory should be updated so that any documentation or other details points to stable rather than incubator. The chart `version` will, also, need to be incremented.
+1. Finally, delete the chart in the incubator directory in a separate PR referencing the PR where the chart was promoted to stable.
```

Comment:
There are negative repercussions for existing applications that consume incubator charts as dependencies if this were to happen. Here are a couple examples:

1) Dependencies can be broken. If someone is referencing the chart in incubator in the [manner the helm docs describe](https://docs.helm.sh/developing_charts/#managing-dependencies-with-requirements-yaml) that other chart will then have a broken requirement because this one is gone.

2) If someone else comes along later and creates a new chart with the same name in incubator it can break apps. Another way to reference a chart as a requirement is by full url. If the old incubator chart and the new had the same name and version the new one would replace the old one. Apps consuming the old one at deploy time would now consume the new one and we would have introduced a silent error.

We don't want to change dependencies out from under people. This would be similar to the famous left-pad incident to do that.

Reply:
> Dependencies can be broken. If someone is referencing the chart in incubator in the manner the helm docs describe that other chart will then have a broken requirement because this one is gone.

AFAIK this won't actually happen. The CI will not go an delete entries in the repo index for removed charts. Correct me if I'm wrong. The second concern is more the worry here.

Reply:
I thought I remember some discussion that deprecated charts should be removed, but maybe I'm wrong. I don't like deprecated charts to stay around forever. What if we create a file that tracks removal of deprecated charts? CI could then check this file when a new chart is added with the name of a deprecated one.